app/models/battle.rb

```diff
@@ -1,16 +1,19 @@
 class Battle
 
-  include MongoMapper::Document
+  include Mongoid::Document
 
-  key :name,      String, required: true
-  key :usernames, Array,  required: true
+  validates :name, :usernames, presence: true
 
-  many :turns
+  field :name, type: String
+  field :usernames, type: Array, default: []
+
+  has_many :turns
 
   def self.wage(map, robots)
     battle = Battle.new name: map.name, usernames: robots.map(&:username)
     turn = Turn.new
     turn.board = Board.draw map, robots
+    turn.battle = battle
```

Comment:
I'm guessing that mongo_mapper did this automatically when you do `battle.turns << turn`, but mongoid doesn't do that.